### DISCUSSION TITLE
BatchNorm with masked data

### DISCUSSION THREAD

**Discussion by swamidass** [google/flax#2739] [Q&A]
What is the best way to mask data from the batch normalization layer? 

The use case is that my dataset requires the padding, and this padding is variable. I don't want the pad values perverting the running statistics. 

Implementation. It should be possible to add this feature to all normalization layers in flax by allowing _compute_stats to optionally take a mask as input, and having each layer appropriately transform the mask (optional arg) and pass it through an optional mask to _compute_stats.

Right now the only options I see are:

1. flax implements mask input to the library BN later (best)

2. Find another library that offers a flax masked-BN layer (ok)

3. Copy and rewrite code from flax BN to create a new BN layer that handles masking (not ideal)

How are other people solving this problem? What is the right way forward?
